src/constants.js

```diff
@@ -46,15 +46,21 @@ function getDirectory(category: string): string {
   return path.join(userHome, `.${category}`, 'yarn');
 }
 
-function getCacheDirectory(): string {
+function getPreferredCacheDirectories(): Array<string> {
+  const preferredCacheDirectories = [getDirectory('cache')];
+
+  if (process.platform !== 'win32') {
+    preferredCacheDirectories.unshift('/tmp/.yarn-cache');
```

Comment:
Nope. Use [os.tmpdir()](https://nodejs.org/api/os.html#os_os_tmpdir) please. Note that this is also platform independent.